api/konnect/v1alpha1/konnect_cloudgateway_transitgateway_types.go

```diff
@@ -0,0 +1,199 @@
+}
+
+// AWSTransitGateway is the configuration of an AWS transit gateway.
+type AWSTransitGateway struct {
+	// Human-readable name of the transit gateway.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=120
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+	// List of mappings from remote DNS server IP address sets to proxied internal domains, for a transit gateway
+	// attachment.
+	//
+	// +kubebuilder:validation:Optional
+	DNSConfig []TransitGatewayDNSConfig `json:"dns_config,omitempty"`
+	// CIDR blocks for constructing a route table for the transit gateway, when attaching to the owning
+	// network.
+	//
+	// +kubebuilder:validation:Required
+	CIDRBlocks []string `json:"cidr_blocks"`
+	// configuration to attach to AWS transit gateway on the AWS side.
+	//
+	// +kubebuilder:validation:Required
+	AttachmentConfig AwsTransitGatewayAttachmentConfig `json:"attachment_config"`
+}
+
+// AzureTransitGateway is the configuration of an Azure transit gateway.
+type AzureTransitGateway struct {
+	// Human-readable name of the transit gateway.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=120
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+	// List of mappings from remote DNS server IP address sets to proxied internal domains, for a transit gateway
+	// attachment.
+	//
+	// +kubebuilder:validation:Optional
+	DNSConfig []TransitGatewayDNSConfig `json:"dns_config,omitempty"`
+	// configuration to attach to Azure VNET peering gateway.
+	//
+	// +kubebuilder:validation:Required
+	AttachmentConfig AzureVNETPeeringAttachmentConfig `json:"attachment_config"`
+}
+
+// TransitGatewayDNSConfig is the DNS configuration of a tansit gateway.
+type TransitGatewayDNSConfig struct {
+	// Remote DNS Server IP Addresses to connect to for resolving internal DNS via a transit gateway.
+	RemoteDNSServerIPAddresses []string `json:"remote_dns_server_ip_addresses,omitempty"`
+	// Internal domain names to proxy for DNS resolution from the listed remote DNS server IP addresses,
+	// for a transit gateway.
+	DomainProxyList []string `json:"domain_proxy_list,omitempty"`
+}
+
+// AwsTransitGatewayAttachmentConfig is the configuration to attach to a AWS transit gateway.
+type AwsTransitGatewayAttachmentConfig struct {
+	// AWS Transit Gateway ID to create attachment to.
+	//
+	// +kubebuilder:validation:Required
+	TransitGatewayID string `json:"transit_gateway_id"`
+	// Resource Share ARN to verify request to create transit gateway attachment.
+	//
+	// +kubebuilder:validation:Required
+	RAMShareArn string `json:"ram_share_arn"`
+}
+
+// AzureVNETPeeringAttachmentConfig is the configuration to attach to a Azure VNET peering gateway.
+type AzureVNETPeeringAttachmentConfig struct {
+	// Tenant ID for the Azure VNET Peering attachment.
+	//
+	// +kubebuilder:validation:Required
+	TenantID string `json:"tenant_id"`
```

Comment:
Can we make each field's godoc start with the field name (as it's idiomatically done)?

e.g. 

```suggestion
	// TenantID is the ID for the Azure VNET Peering attachment.
	//
	// +kubebuilder:validation:Required
	TenantID string `json:"tenant_id"`
```

Reply:
Updated. The definitions are copied from sdk-konnect-go repo and I changed the comments in the latest commit.